tests/suites/test_suite_hmac_drbg.function

```diff
@@ -199,8 +200,9 @@ void hmac_drbg_no_reseed( int md_alg, data_t * entropy,
     TEST_ASSERT( mbedtls_hmac_drbg_random_with_add( &ctx, my_output, output->len,
                                             add2->x, add2->len ) == 0 );
 
-    /* clear for second run */
+    /* reset context for second run */
     mbedtls_hmac_drbg_free( &ctx );
+    mbedtls_hmac_drbg_init( &ctx );
```

Comment:
`hmac_drbg_no_reseed()` test was failing before this change due to the `reseed_interval`  not set to a non zero value after `hmac_drbg_free()`. This caused the subsquent `mbedtls_hmac_drbg_random_with_add()` to reseed, failing the test.

@ronald-cron-arm I would argue that we should not support using the DRBG after `free()` but before another `init()`, since `free()` destroys the context mutex. However we don't say this anywhere in the documentation. What do you think?

Reply:
Sorry for this late reply. I've checked with @gilles-peskine-arm and in fact the test is right and thus shouldn't be changed. mbedtls_hmac_drbg_free() should reset the context to a state that's equivalent to its state after the initial call to mbedtls_hmac_drbg_init(). One could argue that is more a reset than a free probably but that's the convention across mbedtls interfaces. The same apply to ctr_drbg. If this is ok with you, while at it, you are welcome to update the documentation of the `free()` functions to clarify their behavior. It would be much appreciated but no worries if you don't have time to do it.

Reply:
I think this 'reset' convention does not apply to the MBEDTLS_THREADING_C mutex as after `mbedtls_hmac_drbg_free` the mutex handle is NULL. I added `mbedtls_mutex_init` after `mbedtls_platform_zeroize` in `mbedtls_hmac_drbg_free` in the latest patch. Likewise with `ctr_drbg`. This would be consistent with 'resetting context to state after initial call to `mbedtls_hmac_drbg_init`'